routers/install/install.go

```diff
@@ -100,19 +110,18 @@ func Install(ctx *context.Context) {
 	form.DbSchema = setting.Database.Schema
 	form.Charset = setting.Database.Charset
 
-	var curDBOption = "MySQL"
-	switch setting.Database.Type {
-	case "postgres":
-		curDBOption = "PostgreSQL"
-	case "mssql":
-		curDBOption = "MSSQL"
-	case "sqlite3":
-		if setting.EnableSQLite3 {
-			curDBOption = "SQLite3"
+	curDBType := setting.Database.Type
+	var isCurDBTypeSupported bool
+	for _, dbType := range setting.SupportedDatabaseTypes {
+		if dbType == curDBType {
+			isCurDBTypeSupported = true
+			break
 		}
 	}
-
-	ctx.Data["CurDbOption"] = curDBOption
+	if !isCurDBTypeSupported {
```

Comment:
instead of overwrite if wrong why not fail and show an error?

Reply:
That's the old logic.

And here is the install page, we can just show user a default install page(with default database option) instead of showing misleading messages.